integration-tests/utils/integration-utils.js

```diff
@@ -181,24 +183,53 @@ const delay = ms => {
 }
 
 const dismissNotifications = async page => {
-  await delay(3000) // delayed for any alerts to show
-  const notificationCloseButtons = await page.$x(
-    '(//a | //*[@role="button"] | //button)[contains(@aria-label,"Dismiss") and not(contains(@aria-label,"error"))]')
-
+  const notificationCloseButtonXpath = '(//a | //*[@role="button"] | //button)[contains(@aria-label,"Dismiss") and not(contains(@aria-label,"error"))]'
+  // Don't sleep for 3 seconds if a notification was found already
+  await Promise.race([
+    delay(3000), // delayed for any alerts to show
+    page.waitForXPath(notificationCloseButtonXpath, { visible: true })
+  ])
+  const notificationCloseButtons = await page.$x(notificationCloseButtonXpath)
   await Promise.all(
-    notificationCloseButtons.map(handle => handle.click())
+    notificationCloseButtons.map(button => button.evaluateHandle(btn => btn.click(), button))
   )
-
   return !!notificationCloseButtons.length && delay(1000) // delayed for alerts to animate off
 }
 
+const dismissNPSSurvey = async page => {
+  let element
+  try {
+    element = await page.waitForXPath('//iframe[@aria-label="NPS Survey"]', { timeout: 1000 })
+  } catch (e) {
+    return // NPS survey was not found
+  }
+  try {
+    console.log('dismissing NPS survey')
+    const iframe = await element.contentFrame()
+    const [closeButton] = await iframe.$x('.//*[@id="ask-me-later"]') // Find "Ask Me Later" button to dismiss survey popup
+    await closeButton.evaluateHandle(button => button.click(), closeButton)
+    await delay(500) // delayed for survey to animate off
+  } catch (e) {
+    console.error(e)
```

Comment:
Log here because it would be easy to find them in (middle of somewhere) test log file instead finding them at the end of log file in test Summary.
For the second question, I don't know if error would be caught by the `page.pageerror`.

Reply:
Doesn't the `page.pageerror` handler log errors when they occur? Why would this message be in a different place in the logs than the "dismissing NPS survey" message?